sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzePartitionCommand.scala

```diff
@@ -75,6 +75,9 @@ case class AnalyzePartitionCommand(
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val sessionState = sparkSession.sessionState
+    if (sessionState.catalog.getTempView(tableIdent.identifier).isDefined) {
+      throw new AnalysisException("ANALYZE TABLE is not supported on a temporary view.")
+    }
```

Comment:
`ANALYZE TABLE` supports both table and permanent view (if cached), but not a temporary view. Thus, we need to check this in commands to make sure temporary views are resolved first. Another approach is to introduce `UnresolvedTableOrPermanentView` such that if a temporary view is found first, it fails at the `Analyzer`. WDYT @cloud-fan?

Reply:
Seems it's not the first time to hit this requirement. How about we add `allowTemp: Boolean` in `UnresolvedTableOrView`?

Reply:
OK. What if we want to support the scenario where a temp view is allowed but not a permanent view (not this PR)? Do we want something like the following?
```scala
case class UnresolvedTableOrView(
    multipartIdentifier: Seq[String],
    allowTempView: Boolean = true,
    allowPermanentView: Boolean = true) extends LeafNode {
  require(allowTempView || allowPermanentView)
}
```

Reply:
SGTM